src/glfw-wrapper.cpp

```
@@ -0,0 +1,54 @@
+#include "glfw-wrapper.h"
+
+namespace mogl
+{
+    void GLFWWrapper::initGLFW()
+    {
+        if (!glfwInit())
+        {
+            std::cout << "Failed to initialize GLFW.\n";
+            std::exit(EXIT_FAILURE);
```

It's better to throw an exception here, rather than printing to console and exiting. It's violating single responsibility principle, because this method became responsible for init (as name suggests), logging and stopping the application

---

Would you suggest creating a new exception class which inherits from `std::exception`? Or throw some internal exception?

---

Possibly `throw std::runtime_error`?

---

In my opinion, it's always a good solution to create a custom exception. I would inherit it from `std::runtime_error` for convenience, because I'll can have a constructor like that:
```cpp
class initialization_exception : public std::runtime_error {
  using std::runtime_error::runtime_error;
}
```

This trick helps not to write again all of the constructors) But, of course, if you'll need some custom logic here, trick will be useless.